### PR TITLE
Disallow illegal characters in video titles

### PR DESCRIPTION
**Pull request by GrahamGoudeau**
Need some feedback about exactly the right characters to allow. Right now I have it as alphanumeric + spaces.

### CODE REVIEW

**Comment by GrahamGoudeau** [comment]
Fixes #89

File: app/client/components/Upload.tsx [hunk 1]
```diff
@@ -189,7 +189,12 @@ export class Upload extends React.Component<UploadProps, UploadState> {
     }
 
     private handleTitle(e) {
-        this.setOk("videoTitle");
+        const title: string = e.target.value;
+        if (VIDEO_TITLE_RE.test(title)) {
+            this.setOk("videoTitle");
+        } else {
+            this.setErr("videoTitle", "error", "Video titles must contain letters, numbers, and spaces.");
```
Comment (ethanpailes) [inline app/client/components/Upload.tsx]: "must contain only letters, numbers, and spaces" might make it a little clearer.

File: app/client/ts/util.ts [hunk 2]
```diff
@@ -19,6 +19,9 @@ export const safeClient: SafeClient =
 
 export const WATCH_URL_RE: RegExp = /\/watch\/([a-zA-Z0-9\/\+]+)/;
 
+// TODO: does not handle international characters
+export const VIDEO_TITLE_RE: RegExp = /^[A-Za-z0-9 ]+$/;
```
Comment (ethanpailes) [inline app/client/ts/util.ts:23]: http://www.regular-expressions.info/unicode.html#prop

It is probably worth testing it out on regex101.com. We could also just have a blacklist of evil chars instead, but I think white-lists are much better for security.

File: app/client/components/Upload.tsx (hunk 1 above)
Comment (nicholasyan) [reply app/client/components/Upload.tsx]: or alternatively "Videos can only contain letters, numbers, and spaces."

**Review by nicholasyan — COMMENTED** [review]
Otherwise, looks good to me.

File: app/client/ts/util.ts (hunk 2 above)
Comment (Abdisalan) [reply app/client/ts/util.ts:23]: plus emojis?

**Pull request merged**